### PR TITLE
Vb 5421 display correct names for needs review page

### PR DESCRIPTION
**Pull request by adaviesMOJ**
## What does this PR do?
Instead of showing NOT_KNOWN for public bookers and making calls
to the manage user service when not needed, now we determine what
user type the user is and display the correct full name for them.

Staff -> their full name
Public -> "Public User"
System -> "NOT_KNOWN"

## JIRA Link
https://dsdmoj.atlassian.net/browse/VB-4521

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/ManageUsersService.kt [hunk 1]
```diff
@@ -81,4 +84,12 @@ class ManageUsersService(
       manageUsersApiClient.getUserDetails(userName).block(apiTimeout)?.fullName ?: userNameIfNotAvailable
     }
   }
+
+  fun getFullNameFromActionedBy(actionedByDto: ActionedByDto): String {
+    return when (actionedByDto.userType) {
+      UserType.STAFF -> manageUsersApiClient.getUserDetails(actionedByDto.userName!!).block(apiTimeout)?.fullName ?: NOT_KNOWN
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/ManageUsersService.kt:90]: should we use getUserFullName method instead (line 80)?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/ManageUsersService.kt [hunk 2]
```diff
@@ -73,6 +75,7 @@ class ManageUsersService(
     }
   }
 
+  // TODO: Public user calling manage-user-service???
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/ManageUsersService.kt:78]: redundant TODO?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/ManageUsersService.kt (hunk 1 above)
Comment (adaviesMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/ManageUsersService.kt:89]: I've opted away from this as it's used in a few other places that don't have access to the ActionedByDto. They pass directly a String. Avoiding a big refactor by introducing a new method.

**Review by dhirajshettymoj — DISMISSED** [review]
LGTM

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**